src/approaches/grammar_search_invention_approach.py

```diff
@@ -160,13 +160,32 @@ def _check_goal(s: FrozenSet[Predicate]) -> bool:
             del s  # unused
             return False
 
-        # Start the search with all of the candidates.
-        init = frozenset(candidates)
+        if CFG.grammar_search_direction == "largetosmall":
+            # Successively consider smaller predicate sets.
+            def _get_successors(s: FrozenSet[Predicate]
+                    ) -> Iterator[Tuple[None, FrozenSet[Predicate], float]]:
+                for predicate in sorted(s):  # sorting for determinism
+                    # Actions not needed. Frozensets for hashing.
+                    yield (None, frozenset(s - {predicate}), 1.)
+
+            # Start the search with all of the candidates.
+            init = frozenset(candidates)
+        else:
+            assert CFG.grammar_search_direction == "smalltolarge"
+            # Successively consider larger predicate sets.
+            def _get_successors(s: FrozenSet[Predicate]
+                    ) -> Iterator[Tuple[None, FrozenSet[Predicate], float]]:
+                for predicate in sorted(set(candidates) - s):  # determinism
+                    # Actions not needed. Frozensets for hashing.
+                    yield (None, frozenset(s | {predicate}), 1.)
```

Comment:
why are the costs always 1 instead of being the cost of removing/adding that predicate?

Reply:
i mean: the cost of the predicate being added/removed, as produced by generate()

Reply:
this is just like a "prior" on the predicate. those costs are factored in via true positives and false positives. is that what you mean?